packages/input/Input.test.tsx

```diff
@@ -1,48 +1,155 @@
-import { Input } from './Input';
-
-import { h, mount } from 'bore';
-import { emit } from 'skatejs';
+import { h } from '../_helpers';
+import { mount } from 'bore';
+import * as expect from 'expect';
+import { Input } from './index';
 
 describe( Input.is, () => {
 
-  it( 'should set value and emit on change', ( done ) => {
+  describe( `Custom element`, () => {
+    it( `should be registered`, () => {
+
+      expect( customElements.get( Input.is ) ).toBe( Input );
 
-    mount( <Input value="initial" /> ).wait(( element ) => {
+    });
 
-      element.node.addEventListener( 'change', () => {
+    it( `should render via JSX IntrinsicElement`, () => {
 
-        expect(( element.node as HTMLInputElement ).value ).to.equal( 'ok' );
+      return mount(
+        <bl-input value="inputValue" />
+      ).wait(( element ) => {
 
-        done();
+        expect( element.node.localName ).toBe( Input.is );
 
       });
 
-      const innerInput = element.one( 'input' ).node as HTMLInputElement;
-      innerInput.value = 'ok';
-      emit( innerInput, 'change', { composed: false });
-
     });
 
+    it( `should render via JSX class`, () => {
+
+      return mount(
+        <Input value="inputValue" />
+      ).wait(( element ) => {
+
+        expect( element.has( '.c-field' ) ).toBe( true );
+
+      });
+
+    });
   });
 
-  it( 'should set value and emit on input', ( done ) => {
+  describe( `API`, () => {
```

Comment:
Yes, when you have one minute of your time, please come here and we will create spec for it.